docs/getting-started-guides/mesos.md

```diff
@@ -0,0 +1,303 @@
+## Getting started with Kubernetes on Mesos
+
+<!-- TODO: Update, clean up. -->
+
+Mesos allows dynamic sharing of cluster resources between Kubernetes and other first-class Mesos frameworks such as [Hadoop][1], [Spark][2], and [Chronos][3].
+Mesos ensures applications running on your cluster are isolated and that resources are allocated fairly.
+
+Running Kubernetes on Mesos allows you to easily move Kubernetes workloads from one cloud provider to another to your own physical datacenter.
+
+This tutorial will walk you through setting up Kubernetes on a Mesos cluster on [Google Cloud Plaform][4].
+It provides a step by step walk through of adding Kubernetes to a Mesos cluster and running the classic GuestBook demo application.
+The walkthrough presented here is based on the v0.4.x series of the Kubernetes-Mesos project, which itself is based on Kubernetes v0.11.0.
+
+### Prerequisites
+
+* Understanding of [Apache Mesos][11]
+* Mesos cluster on [Google Compute Engine][5]
+* Identify the Mesos master node external IP from Mesosphere [cluster launch pad][12]
+* A [VPN connection to the cluster][6], or setup firewall rules for direct access (used in this doc).
+
+### Setup Access to Mesos Admin GUI
+This doc will refer to the running Mesos cluster repeatedly.  It's a good idea to make the Mesos admin GUI accessible via the browser.  First, take note of the network name attached to your Mesos cluster (either via the Google Cloud GUI) or with `gcloud` tool as follows:
+
+```bash
+gcloud compute firewall-rules list
+```
+Next, add a firewall-rule to open traffic on port 5050.  This will allow you to access Mesos master GUI via the browser.
+
+```bash
+gcloud compute firewall-rules create mesos-master-gui --network ${network_name}  --source-ranges 0.0.0.0/0 --allow tcp:5050
+```
+Once the rule is applied, point your browser to your Mesos master's IP address (see pre-requisites).  You should see the Mesos GUI displayed.
```

Comment:
I may have misunderstood, but when you said "setup firewall rules for direct access" earlier, I took it to mean I could access http://<external IP of my master>:5050. But this doesn't work. Here's how mesos-master is running (output of ps below). I didn't try the VPN approach yet but probably that does work, accessing the internal 10.x address.

```
root      2454  0.3  0.3 366984 24052 ?        Sl   07:08   0:06 /usr/sbin/mesos-master --zk=zk://10.188.173.32:2181/mesos --port=5050 --log_dir=/var/log/mesos --cluster=development-5384 --hostname=10.188.173.32 --ip=10.188.173.32 --quorum=1 --registry=in_memory --work_dir=/var/lib/mesos
```